javav2/example_code/inspector/src/main/java/com/java/inspector/InspectorScenario.java

```diff
@@ -0,0 +1,179 @@
+
+        // Step 1
+        logger.info(DASHES);
+        logger.info("Step 1: Checking Inspector account status...");
+        String status = actions.getAccountStatusAsync().join();
+        logger.info(status);
+        waitForInputToContinue();
+
+        // Step 2
+        logger.info(DASHES);
+        logger.info("Step 2: Enabling Inspector...");
+        String message = actions.enableInspectorAsync(null).join();
+        logger.info(message);
+        waitForInputToContinue();
+
+        // Step 3
+        logger.info(DASHES);
+        logger.info("Step 3: Listing findings...");
+        String allFindings = actions.listLowSeverityFindingsAsync().join();
+
+        if (!allFindings.equals("No findings found.")) {
+            // Split by newline and get the last ARN
+            String[] arns = allFindings.split("\\r?\\n");
+            String lastArn = arns[arns.length - 1];
+
+            // Looks up details
+            logger.info("Look up details on: {}" , lastArn);
+            waitForInputToContinue();
+            String details = actions.getFindingDetailsAsync(lastArn).join() ;
+            logger.info(details);
+        } else {
+            System.out.println("No findings found.");
+        }
+
+        waitForInputToContinue();
+
+        // Step 4
+        logger.info(DASHES);
+        logger.info("Step 4: Listing coverage...");
+        String coverage = actions.listCoverageAsync(5).join();
+        logger.info(coverage);
+        waitForInputToContinue();
+
+        // Step 5
+        logger.info(DASHES);
+        logger.info("Step 5: Creating filter...");
+        String filterName = "suppress-low-" + System.currentTimeMillis();
+        String filterArn = actions
+                .createLowSeverityFilterAsync(filterName, "Suppress low severity findings")
+                .join();
+        logger.info("Created filter: {}", filterArn);
+        waitForInputToContinue();
+
+        // Step 6
+        logger.info(DASHES);
+        logger.info("Step 6: Listing filters...");
+        String filters = actions.listFiltersAsync(10).join();
+        logger.info(filters);
+        waitForInputToContinue();
+
+        // Step 7
+        logger.info(DASHES);
+        logger.info("Step 7: Usage totals...");
+        String usage = actions.listUsageTotalsAsync(null, 10).join();
+        logger.info(usage);
+        waitForInputToContinue();
+
+        // Step 8
+        logger.info(DASHES);
+        logger.info("Step 8: Coverage statistics...");
+        String stats = actions.listCoverageStatisticsAsync().join();
+        logger.info(stats);
+        waitForInputToContinue();
+
+        // Step 9
+        logger.info(DASHES);
+        logger.info("Step 9: Delete filter?");
+        logger.info("Filter ARN: {}", filterArn);
+        logger.info("Delete the filter and disable Inspector? (y/n)");
+
+        if (scanner.nextLine().trim().equalsIgnoreCase("y")) {
+            actions.deleteFilterAsync(filterArn).join();
```

Comment:
The main scenario is missing the outer try/catch that will run the cleanup on a failure.